src/custom/hooks/useWalletInfo.ts

```diff
@@ -60,7 +60,8 @@ function checkIsSupportedWallet(params: {
 async function getWcPeerMetadata(connector: WalletConnectConnector): Promise<{ walletName?: string; icon?: string }> {
   const provider = (await connector.getProvider()) as WalletConnectProvider
 
-  const meta = provider.walletMeta
+  // fix for this https://github.com/gnosis/cowswap/issues/1929
+  const meta = provider.walletMeta || provider.signer.connection.wc.peerMeta
```

Comment:
cant this just be provider.peerMeta? why the long chain in signer?

Reply:
Also from Nenad's original PR.
Anyway, I'll update with your suggestion